generator/wrapper_gen.py

```diff
@@ -175,7 +175,7 @@ def my_procid():
 
 def _get_data(x):
     if isinstance(x, pdDataFrame):
-        x = [x.loc[:,i].values for i in x]
+        x = x.values
```

Comment:
In most cases this will make a copy if the DF is not contiguous. This basically disables the SOA path.